uwrt_mars_rover_hw/include/uwrt_mars_rover_hw/uwrt_mars_rover_hw_drivetrain.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <hardware_interface/controller_info.h>
+#include <hardware_interface/joint_command_interface.h>
+#include <hardware_interface/joint_state_interface.h>
+#include <hardware_interface/robot_hw.h>
+
+namespace uwrt_mars_rover_hw {
+
+class UWRTRoverHWDrivetrain : public hardware_interface::RobotHW {
+ public:
+  explicit UWRTRoverHWDrivetrain();
+
+  struct DrivetrainJointState {
+    double position;
+    double velocity;
+    double effort;
+  };
+
+  struct DrivetrainJointCommand {
+    enum class Type { NONE, POSITION, VELOCITY, EFFORT };
+    Type type;
+    double data;
+  };
+
+  bool init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw_nh) override;
```

Comment:
should this be virtual too? i'm dumb tho so idk

Reply:
nah, override already means theres a virtual function in the base class so while virtual can be added here, it is redundant. in fact, its a clang tidy waarning lol